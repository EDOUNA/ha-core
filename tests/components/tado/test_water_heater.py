"""The water heater tests for the Tado platform."""

from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant

from .util import async_init_integration


async def test_water_heater_create_sensors(
    hass: HomeAssistant, snapshot: SnapshotAssertion
) -> None:
    """Test creation of water heater."""

    await async_init_integration(hass)

    state = hass.states.get("water_heater.water_heater")
    assert state.state == "auto"

    # Only test for a subset of attributes in case
    # HA changes the implementation and a new one appears
    expected_keys = [
        "current_temperature",
        "friendly_name",
        "max_temp",
        "min_temp",
        "operation_list",
        "operation_mode",
        "supported_features",
        "target_temp_high",
        "target_temp_low",
        "temperature",
    ]

    actual_attributes_subset = {key: state.attributes.get(key) for key in expected_keys}
    assert actual_attributes_subset == snapshot

    state = hass.states.get("water_heater.second_water_heater")
    assert state.state == "heat"

    # Only test for a subset of attributes in case
    # HA changes the implementation and a new one appears
    expected_keys = [
        "current_temperature",
        "friendly_name",
        "max_temp",
        "min_temp",
        "operation_list",
        "operation_mode",
        "supported_features",
        "target_temp_high",
        "target_temp_low",
        "temperature",
    ]

    actual_attributes_subset = {key: state.attributes.get(key) for key in expected_keys}
    assert actual_attributes_subset == snapshot
