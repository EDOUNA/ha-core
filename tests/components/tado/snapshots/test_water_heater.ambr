# serializer version: 1
# name: test_water_heater_create_sensors
  dict({
    'current_temperature': None,
    'friendly_name': 'Water Heater',
    'max_temp': 31.0,
    'min_temp': 16.0,
    'operation_list': list([
      'auto',
      'heat',
      'off',
    ]),
    'operation_mode': 'auto',
    'supported_features': <WaterHeaterEntityFeature: 3>,
    'target_temp_high': None,
    'target_temp_low': None,
    'temperature': 65.0,
  })
# ---
# name: test_water_heater_create_sensors.1
  dict({
    'current_temperature': None,
    'friendly_name': 'Second Water Heater',
    'max_temp': 31.0,
    'min_temp': 16.0,
    'operation_list': list([
      'auto',
      'heat',
      'off',
    ]),
    'operation_mode': 'heat',
    'supported_features': <WaterHeaterEntityFeature: 3>,
    'target_temp_high': None,
    'target_temp_low': None,
    'temperature': 30.0,
  })
# ---
